lib/instrumental_tools/system_inspector/linux.rb

```diff
@@ -0,0 +1,145 @@
+class SystemInspector
+  module Linux
+    def self.load_cpu
+      output = { :gauges => {} }
+      output[:gauges].merge!(cpu)
+      output[:gauges].merge!(loadavg)
+      output
+    end
+
+    def self.cpu
+      categories = [:user, :nice, :system, :idle, :iowait]
+      values = `cat /proc/stat | grep cpu[^0-9]`.chomp.split(/\s+/).slice(1, 5).map { |v| v.to_f }
+      SystemInspector.memory.store(:cpu_values, values.dup)
+      if previous_values = SystemInspector.memory.retrieve(:cpu_values)
+        index = -1
+        values.collect!{ |value| (previous_values[index += 1] - value).abs }
+      end
+      data = Hash[*categories.zip(values).flatten]
+      total = values.inject { |memo, value| memo + value }
+
+      output = {}
+      if previous_values
+        data.each do |category, value|
+          output["cpu.#{category}"] = value / total * 100
+        end
+      end
+      output["cpu.in_use"] = 100 - data[:idle] / total * 100
+      output
+    end
+
+    def self.loadavg
+      min_1, min_5, min_15 = `cat /proc/loadavg`.split(/\s+/)
+      {
+        'load.1min' => min_1.to_f,
+        'load.5min' => min_5.to_f,
+        'load.15min' => min_15.to_f,
+      }
+    end
+
+    def self.load_memory
+      output = { :gauges => {} }
+      if SystemInspector.command_present?('free', 'memory')
+        output[:gauges].merge!(memory)
+      end
+      if SystemInspector.command_present?('free', 'swap')
+        output[:gauges].merge!(swap)
+      end
+      output
+    end
+
+    def self.memory
+      _, total, used, free, shared, buffers, cached = `free -k -o | grep Mem`.chomp.split(/\s+/)
+      {
+        'memory.used_mb' => used.to_f / 1024,
```

Comment:
Time to monkey patch. O_O

No, but maybe a method for `to_megabytes` or something that these bit all use.

Or maybe just a constant.